app/user/admin.py

```diff
@@ -0,0 +1,91 @@
+from flask import render_template, redirect, url_for, flash, abort
+from flask_login import login_required, current_user
+from flask_wtf import FlaskForm
+from wtforms import StringField, SubmitField, PasswordField
+from wtforms.validators import DataRequired, Email, EqualTo, Optional
+from wtforms.fields.html5 import EmailField
+from app.views import confirm, is_safe_url
+from . import groups_required, user_blueprint
+from .models import User
+
+class UserEditForm(FlaskForm):
+    username = StringField('Username')
+    givenName = StringField('Given Name', validators=[DataRequired('Please enter a given name')])
+    surname = StringField('Surname', validators=[DataRequired('Please enter a surname')])
+    mail = EmailField('E-Mail', validators=[
+            DataRequired('Please enter an E-Mail address'),
+            Email('Please enter a valid E-Mail address')])
+    password = PasswordField('Password', validators=[
+            Optional(False),
+            EqualTo('confirm_password', 'Passwords must match')])
+    confirm_password = PasswordField('Repeat password')
+    submit = SubmitField('Submit')
+
+@user_blueprint.route('/admin/users')
+@login_required
+@groups_required('admin')
+def list_users():
+    users = User.query()
+    return render_template('admin/listUsers.html',
+        users = users
+    )
+
+@user_blueprint.route('/admin/user/<string:username>/delete', methods=['GET', 'POST'])
+@login_required
+@confirm(title='Delete User?',
+        prompt='Are you sure you want to delete this user?',
+        action='Delete',
+        back='user.list_users')
+@groups_required('admin')
+def delete_user(username):
+    user = User.get(username)
+    if not user:
+        flash('Invalid user name!', 'error')
+        return redirect(url_for('user.list_users'))
+    user.delete()
+    flash('User has been deleted', 'success')
+    return redirect(url_for('user.list_users'))
+
+@user_blueprint.route('/admin/user/<string:username>/edit', methods=['GET', 'POST'])
+@login_required
+def edit_user(username, back_url = None):
+    if not back_url or not is_safe_url(back_url):
```

Comment:
minor nitpick here: applying De Morgan's theorem once would lead to a more readable statement:
```python
if not (back_url and is_safe_url(back_url)):
```

Actually it might make sense to just make is_safe_url return `False` in this case since `None` is obviously not a safe URL.